### DISCUSSION TITLE
Remove dependencies that does not support React Native

### DISCUSSION THREAD

**Discussion by madsbuch** [jotaijs/jotai-devtools#41] [General]
We just had to stop using `jotai-devtools` as it includes a dependency on the https://mantine.dev/ framework.

Generally, it would be nice if packages like this one were headless and do not expect a document or to be run in a browser.

Anyways, nice work and thanks for the time we got to have together!

**Comment by arjunvegda** [comment]
Sorry to hear that! I'd love to learn more about your issues and see if we could help fix those.

Could you elaborate a little on the error/problem you're seeing? Are you using the new `<DevTools/>` or the dev tools hooks?

**Comment by madsbuch** [comment]
The main issue is that Mantine assumes that the `document` property is available, which it is not in react native.

We previously used a structure like this:

```
useAtomsDevtools("GlobalStore", {
     store: store.instance,
});
```

In particular, our usage pattern, as is evident, does not keep everything in the react world, and we greatly appreciate the first-class store support!

**Reply by arjunvegda** [reply]
We may have found a workaround that you might be interested in 🙌 . Please see: https://github.com/jotaijs/jotai-devtools/issues/50#issuecomment-1480486726

**Comment by arjunvegda** [comment]
So would importing the `useAtomsDevtools` from a separate entry (`/utils`) without a reference to `@mantine/*` fix the issue?

Could you try `import { useAtomsDevtools } from 'jotai-devtools/utils'`?

**Reply by Sly777** [reply]
I tried this as well but then It gives this error

```
iOS Bundling failed 1280ms
Unable to resolve "jotai-devtools/utils" from "src/states/store.ts"
```

this is the usage - I'm using expo latest (48.0.9) and react native 0.71.4

```
import { atom, createStore } from 'jotai';
import { useAtomsDevtools } from 'jotai-devtools/utils';
import { ReactElement } from 'react';

const appStore = createStore();

const countAtom = atom(0);
appStore.set(countAtom, 1);
const unsub = appStore.sub(countAtom, () => {
  console.log('countAtom value is changed to', appStore.get(countAtom));
});

export const AtomsDevtools = ({
  children,
}: {
  children: ReactElement;
}): ReactElement => {
  useAtomsDevtools('appstore');
  return children;
};

export default appStore;
```

```
<Provider store={appStore}>
      <AtomsDevtools>
         <Stack />
      </AtomsDevtools>
    </Provider>
```

**Reply by arjunvegda** [reply]
That looks like a bug! Thanks for the snippets, and other info, it is very helpful! 

I'll look into it soon, logged the #50 issue for tracking purposes. 

~Workaround for now on your end could be to downgrade to `jotai-devtools@0.2.0`~

See [my comment](https://github.com/jotaijs/jotai-devtools/issues/50#issuecomment-1480486726)

**Comment by Sly777** [comment]
Is there any update on this? Basically because of `@mantine` package, devtool doesn't work on react-native (even with `useAtomsDevtools` func usage only)

Error log;
```
iOS Bundling failed 3195ms
Unable to resolve "react-dom" from "node_modules/@mantine/core/cjs/Portal/Portal.js"
```

**Reply by arjunvegda** [reply]
Let's continue our discussion [here](https://github.com/jotaijs/jotai-devtools/issues/50#issuecomment-1480486726)
